File: 1. Arrays & Strings/8_five_sort/8_five_sort.py
def five_sort(array):
    start_index = 0
    end_index = len(array) - 1

    while start_index < end_index:
        if array[end_index] == 5:
            # 5 at the right no need to swap move right pointer to left
            end_index -= 1
        elif array[start_index] != 5:
            # not a 5 at the left so no need to swap, move left pointer to right
            start_index += 1
        else:
            array[start_index], array[end_index] = array[end_index], array[start_index]
    
    return array

File: 1. Arrays & Strings/8_five_sort/8_five_sort.java
import java.util.List;
import java.util.Collections;

class Source {
    public static List<Integer> fiveSort(List<Integer> array) {
        int startIndex = 0;
        int endIndex = array.size() - 1;

        while (startIndex < endIndex) {
            if (array.get(endIndex) == 5) {
                // 5 at the right no need to swap move right pointer to left
                endIndex--;
            } else if (array.get(startIndex) != 5) {
                // not a 5 at the left so no need to swap, move left pointer to right
                startIndex++;
            } else {
                Collections.swap(array, startIndex, endIndex);
            }
        }

        return array;
    }
}